lib/runnable.js

```diff
@@ -439,3 +380,48 @@ Runnable.prototype._timeoutError = function(ms) {
   }
   return new Error(msg);
 };
+
+/**
+ * Mixes in a Behavior to this Runnable or returns current.
+ * @param {Object} [behavior] - Behavior mixin. If unset, returns current Behavior
+ * @public
+ * @returns {Object|Runnable}
+ */
+Runnable.prototype.behavior = function(behavior) {
+  if (arguments.length) {
+    var currentBehavior = Object.assign({}, this._behavior);
+    Object.keys(behavior).forEach(function(behaviorName) {
+      currentBehavior[behaviorName] = Object.assign(
+        {},
+        currentBehavior[behaviorName] || {},
+        behavior[behaviorName]
+      );
+    });
+    this._behavior = currentBehavior;
+    return this;
+  }
+  return this._behavior;
+};
+
+/**
+ * Calls a behavior method by name with array args, `Function.prototype.apply` style
+ * @param {string} name - Behavior method name
+ * @param {*[]} [args] - Array of arguments
+ * @private
+ * @returns {*} Whatever the behavior method returns
+ */
+Runnable.prototype.applyBehavior = function(name, args) {
+  return this.behavior().Runnable[name].apply(this, args || []);
+};
+
+/**
+ * Calls a behavior method by name with positional args, `Function.prototype.call` style
+ * @param {string} name - Behavior method name
+ * @param {...*} [arg] - Arguments
+ * @private
+ * @returns {*} Whatever the behavior method returns
+ */
+Runnable.prototype.callBehavior = function(name) {
```

Comment:
not sure about backward compatibility but can't we use `(name, ...args)`?

Reply:
No, Mocha is written in ES5 :smile: